docs/deprecations.md

```diff
@@ -17,6 +17,17 @@ disabledDeprecations:
   - '*' # To disable all deprecation messages
 ```
 
+<a name="NESTED_CUSTOM_CONFIGURATION_PATH'"><div>&nbsp;</div></a>
+
+## Service configurations should not be nested in service sub directories
+
+Deprecation code: `NESTED_CUSTOM_CONFIGURATION_PATH`
```

Comment:
How come this needs to be deprecated and then disabled in v3?

Reply:
@jgeurts Have you read the reasoning?

Reply:
No, I think I missed where it was. Are there changes in v3 that move this beyond confusing for some people?

Reply:
> No, I think I missed where it was

https://www.serverless.com/framework/docs/deprecations#service-configurations-should-not-be-nested-in-service-sub-directories

> Are there changes in v3 that move this beyond confusing for some people?

Sorry, I don't understand that. Can you clarify what you mean?

Reply:
I wasn't sure if this was primarily to avoid confusion for new people setting up projects or if there was a more technical reason that changes in v3 would prevent working with nested config files.

Reply:
> I wasn't sure if this was primarily to avoid confusion for new people setting up projects

It's to avoid confusion. In other words: How properties are resolved in case of nested configuration is not intuitive and leads to user errors. Also it was never meant to be supported, it's supported only due to leaky first implementation.